Add full-text query documentation (#5428)
opensearch-project/documentation-website: _search-plugins/sql/full-text.md

### Syntax

The `MULTI_MATCH` function lets you *boost* certain fields using **^** character. Boosts are multipliers that weigh matches in one field more heavily than matches in other fields. The syntax allows to specify the fields in double quotes, single quotes, surrounded by backticks, or unquoted. Use star ``"*"`` to search all fields. Star symbol should be quoted.

```sql
multi_match([field_expression+], query_expression[, option=<option_value>]*)
```

> The weight is optional and is specified after the field name. It could be delimited by the `caret` character -- `^` or by whitespace. Refer to the following examples:

Reviewer: ```suggestion
The weight is optional and is specified after the field name. It could be delimited by the `caret` character -- `^` or by white space. Refer to the following examples:
```

### Syntax

> The `QUERY_STRING` function has syntax similar to `MATCH_QUERY` and lets you *boost* certain fields using **^** character. Boosts are multipliers that weigh matches in one field more heavily than matches in other fields. The syntax allows to specify the fields in double quotes, single quotes, surrounded by backticks, or unquoted. Use star ``"*"`` to search all fields. Star symbol should be quoted.

Reviewer: ```suggestion
The `QUERY_STRING` function has syntax similar to `MATCH_QUERY` and *boosts* certain fields by using **^** character. Boosts are multipliers that weigh matches in one field more heavily than matches in other fields. The syntax supports specifying the fields with double quotes, single quotes, backticks, or without any quotes. Use star ``"*"`` to search all fields. Star symbol should be quoted.
```

```sql
query_string([field_expression+], query_expression[, option=<option_value>]*)
```

> The weight is optional and is specified after the field name. It could be delimited by the `caret` character -- `^` or by whitespace. Refer to the following examples:

Reviewer: ```suggestion
The weight is optional and is specified after the field name. It could be delimited by the `caret` character -- `^` or by white space. Refer to the following examples:
```

### Syntax

> The syntax allows to specify the fields in double quotes, single quotes, surrounded by backticks, or unquoted. Use star ``"*"`` to search all fields. Star symbol should be quoted.

Reviewer: ```suggestion
The syntax supports specifying the fields with double quotes, single quotes, backticks, or without any quotes. Use star ``"*"`` to search all fields. Star symbol should be quoted.
```

```sql
simple_query_string([field_expression+], query_expression[, option=<option_value>]*)
```

> The weight is optional and is specified after the field name. It could be delimited by the `caret` character -- `^` or by whitespace. Refer to the following examples:

Reviewer: ```suggestion
The weight is optional and is specified after the field name. It could be delimited by the `caret` character -- `^` or by white space. Refer to the following examples:
```

- `analyzer`
- `operator`

> Refer to the `match_bool_prefix` query [documentation]({{site.url}}{{site.baseurl}}/query-dsl/full-text/match-bool-prefix/)  for parameter descriptions and supported values.

Reviewer: ```suggestion
Refer to the `match_bool_prefix` query [documentation]({{site.url}}{{site.baseurl}}/query-dsl/full-text/match-bool-prefix/) for parameter descriptions and supported values.
```
